libraries/AP_NavEKF3/AP_NavEKF3_Measurements.cpp

```diff
@@ -521,8 +521,8 @@ void NavEKF3_core::readGpsData()
                 // and set the corresponding variances and covariances
                 alignMagStateDeclination();
 
-                // Set the height of the NED origin to ‘height of baro height datum relative to GPS height datum'
-                EKF_origin.alt = gpsloc.alt - baroDataNew.hgt;
+                // Set the height of the NED origin
+                ekfGpsRefHgt = (double)0.01 * (double)gpsloc.alt + (double)outputDataNew.position.z;
```

Comment:
Is is necessary because the height could be 10000 metres above sea level for some applications, which introduce rounding errors that prevent a single precision  variable converging when used recursively by the estimator.